code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -80,9 +80,12 @@
 			user << "<span class='danger'>[src] buzzes and beeps.</span>"
 
 /mob/living/simple_animal/bot/cleanbot/process_scan(obj/effect/decal/cleanable/D)
+	. = list()
 	for(var/T in target_types)
 		if(istype(D, T))
-			return D
+			. += D
+	if(.)
+		. = pick(.)

```

Comment:
I know what that means, I am referring to the change he made. How does this cause cleanbots to avoid stepping over each other? It looks like you are just making a list and randomizing the tile selected? When I first made the comment, I thought you were trying to do some sort of shared list among the cleanbots.